PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt

```diff
@@ -103,8 +103,10 @@ class PayPalClient internal constructor(
         payPalRequest: PayPalRequest,
         callback: PayPalPaymentAuthCallback
     ) {
+        analyticsParamRepository.reset()
```

Comment:
Will this cause a problem with ShopperInsights now because the session ID will be different across between the shopper insights analytics and the start of the PayPal flow?

Reply:
Ah good catch. Yeah this will cause issues with ShopperInsights. I think we may need to reset the analyticsParamRepository at the end of every flow. I'll do some investigating and see if this is possible.

The issue i ran into is that the param repository was never cleared, so subsequent launches of the SDK had lingering values for `merchantEnabledAppSwitch` and `payPalServerSideAttemptedAppSwitch`.

Reply:
This was existing, but are we okay with session_id remaining consistent for subsequent launches of the SDK?

I did some testing with resetting the repository at the end of the flow and things seem to be working as expected. Although we need to invoke `reset()` in a number of places (cancel, failure, success cases).